apps/examples/wifi_manager/wm_test/wm_test_stress_test5.c

```diff
@@ -0,0 +1,826 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+#include <tinyara/config.h>
+#include <stdio.h>
+#include <errno.h>
+#include <sys/socket.h>
+#include <sys/time.h>
+#include <sys/types.h>
+#include <wifi_manager/wifi_manager.h>
+#include <stress_tool/st_perf.h>
+#include <netdb.h>
+#include "wm_test.h"
+#include "wm_test_utils.h"
+#include "wm_test_network.h"
+#include "wm_test_log.h"
+#include "wm_test_stress_test5.h"
+
+#define TAG "[WTS5]"
+#define WM_TEST_TRIAL CONFIG_WIFIMANAGER_TEST_TRIAL
+#define WT_UDP_ADDR CONFIG_LWIP_DHCPS_SERVER_IP
+#define WT_UDP_PORT 8888
+#define WT_TCP_PORT 8889 
+#define WT_BUF_SIZE 1024
+//if semaphore operation failed then it'll try it again 10ms later
+#define WT_SEM_TRY_WAIT_US 10000
+#define HOSTNAME_BUFSIZE 128
+#define BROADCASTING_PORT 8890
+
+static char *WM_AP_SSID;
+static char *WM_AP_PASSWORD;
+static wifi_manager_ap_auth_type_e WM_AP_AUTH;
+static wifi_manager_ap_crypto_type_e WM_AP_CRYPTO;
+static char *WM_SOFTAP_SSID;
+static char *WM_SOFTAP_PASSWORD;
+static int WM_SOFTAP_CHANNEL;
+static uint8_t WM_IS_DTLS;
+static uint32_t WM_DTLS_DATA_SIZE;
+static uint8_t WM_IS_TLS;
+static uint32_t WM_TLS_DATA_SIZE;
+static char *WM_HOSTNAME_FILE;
+static sem_t g_wm_sem;
+
+static char recv_mac_buf[16] = {'\0',};
+static char mac_buf[16] = {'\0',};
+static char ipv4_buf[18] = {'\0',};
+static char broadcast_message[30] = {'\0',}; 
+volatile bool gbroadcast_flag;
+
+#define WM_TEST_SIGNAL							\
+	do {								\
+		while (sem_post(&g_wm_sem) != 0) {			\
+			usleep(WT_SEM_TRY_WAIT_US);			\
+		}							\
+		WT_LOG(TAG, "send signal");				\
+	} while (0)
+
+#define WM_TEST_WAIT							\
+	do {								\
+		WT_LOG(TAG, "wait signal");				\
+		while (sem_wait(&g_wm_sem) != 0) {			\
+			if (errno == EINTR) {				\
+				continue;				\
+			}						\
+			WT_LOGE(TAG, "sem wait fail %d", errno);	\
+		}							\
+	} while (0)
+
+/*
+ * callbacks
+ */
+static void wm_sta_connected(wifi_manager_cb_msg_s msg, void *arg);
+static void wm_sta_disconnected(wifi_manager_cb_msg_s msg, void *arg);
+static void wm_softap_sta_join(wifi_manager_cb_msg_s msg, void *arg);
+static void wm_softap_sta_leave(wifi_manager_cb_msg_s msg, void *arg);
+static void wm_scan_done(wifi_manager_cb_msg_s msg, void *arg);
+
+static wifi_manager_cb_s g_wifi_callbacks = {
+	wm_sta_connected,
+	wm_sta_disconnected,
+	wm_softap_sta_join,
+	wm_softap_sta_leave,
+	wm_scan_done,
+};
+
+
+void wm_sta_connected(wifi_manager_cb_msg_s msg, void *arg)
+{
+	WT_LOG(TAG, "--> res(%d)", msg.res);
```

Comment:
wm_sta_connected can return conn_fail event. it doesn't seem to consider the case. Please check

Reply:
Changes done as suggested.